src/features/SpellCheckerProvider.ts

```diff
@@ -202,8 +202,10 @@ export default class SpellCheckerProvider implements vscode.CodeActionProvider
 
 		text = this.processUserIgnoreRegex( text );
 
-		// remove pandoc yaml header
-		text = text.replace( /---(.|\n)*\.\.\./g, ' ' );
```

Comment:
Thanks for the additional information. Since pandoc only contains frontmatter at the beginning of the document, I will change that to being only removed from the beginning of the document. After I get some updates finished in the next week or so, I'll try work on adding a method of turning off some of these regex's.